HungryPhilosophers/src/Fork.java

```diff
@@ -0,0 +1,12 @@
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class Fork {
+	public int id;
+	
+	final Lock lock = new ReentrantLock();
```

Comment:
This approach is:
1) Bracking encapsulation.
2) Forces you to use the reentrantrant lock only. 

What if Java 1.9 introduces more appropriate approach for locks? In this case you will have to rewrite both code for fork and for its user (thankfully for now it is only philosopher). Try to encapsulte locking into Fork methods.